### ISSUE TITLE
Add comment for all Docker added iptables rules so that they can be easily identified

### ISSUE THREAD

**Issue by quickwind** [docker/docker#18908]
This is a proposal to add comment for all Docker added iptables rules so that they can be easily identified, for example as following:

Chain DOCKER (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.3           tcp dpt:5000 /* Added automatically by Docker */

And the comment text can be customized by "--iptables-comment" option for Docker daemon startup. 

This would help a lot in configuration management automation. For example in our case, we are using puppet to manage our environment configuration, because the iptables rules added by Docker are dynamic, we want to preserve it, but puppet firewall module will purge all old rules not defined in puppet, and it is not obvious and simple to define "ignore" for purge, some articles are using:

firewallchain { 'DOCKER:filter:IPv4':
  purge  => true,
  ignore => “172.17.“
}

but it may not be true all the time, if we could have this comment, then it would help a lot for the automation tools to identify docker added rules.

**Comment by thaJeztah** [comment]
Actually, I suggested this in https://github.com/docker/docker/pull/12437#issuecomment-93854813, but there were concerns if the `--comment` option is always available (I think it's an optional module)

**Comment by JrCs** [comment]
+1

**Comment by JrCs** [comment]
But the best will be to have a docker command to dump all the iptables rules that docker create dynamically so they can be reinject after a purge of iptables rules.

**Comment by quickwind** [comment]
Yes, this may not be supported by all iptables versions, but it definitely worth to do, because I believe lots of people using iptables, puppet and Docker, this will make the management a lot of easier...And I doubt old versions percentage in Docker environments...

**Comment by thaJeztah** [comment]
Well, I wasn't sure, because it's not *version* related, but an optional module.

@aboch @mrjana @mavenugo wdyt? Would it be safe so assume that module/option is present?

**Comment by quickwind** [comment]
@thaJeztah I guess it should be a safe assumption as puppet firewall module is using "-m comment" a lot and didn't see any code there to make sure it is present...

**Comment by thaJeztah** [comment]
@quickwind thanks, that's useful information

**Comment by mavenugo** [comment]
Yes. I think it will be useful.
@quickwind pls push a PR in docker/libnetwork.

**Comment by quickwind** [comment]
@mavenugo Ok, I will try but I don't have any experience with Golang, it may take a while...btw, any hint about which file/logic to change:)

**Comment by LK4D4** [comment]
docker/libnetwork#1096 for reference
